### PR TITLE
fix(search-input): propagate search term usage to not found function

### PR DESCRIPTION
**Pull request by DeadEnglish**
## Screenshot / video

https://github.com/marshmallow-insurance/smores-react/assets/5758372/a891aab4-dad2-42b6-80ca-7fbc26d32b25

## What does this do?

Fixing an issue where we could not use the search term entered into the search input component as this is a local state variable.

### CODE REVIEW

File: src/SearchInput/SearchInput.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ export interface SearchInputProps extends CommonFieldProps {
   /**  callback to handle found item click */
   onFound: (element: string) => void
```
Comment (david-gomes5) [inline src/SearchInput/SearchInput.tsx:31]: I know `onFound` was probably created before this work started but it seems like this callback is called when an option is selected?

I initially confused `onFound` with results being found, not selected. I think this prop _should_ be changed to something like `onSelectedOption` or `onSelectChange`. Although, probably not something for this PR (as assuming will be breaking change)

**Review by david-gomes5 — APPROVED** [review]

File: src/SearchInput/components/NoOptions.tsx
Comment (david-gomes5) [inline src/SearchInput/components/NoOptions.tsx:1]: Can we rename this file to from `NoOptions` to `EmptyResults`?

**Review by david-gomes5 — CHANGES_REQUESTED** [review]
`src/SearchInput/components/NoOptions.tsx`

Can we rename this file to from `NoOptions` to `EmptyResults`?

**Review by david-gomes5 — APPROVED** [review]
Just a small request on file name change, otherwise LGTM 👍

File: src/SearchInput/SearchInput.tsx (hunk 1 above)
Comment (DeadEnglish) [reply src/SearchInput/SearchInput.tsx:31]: agreed, I think when we make the breaking change to default fuzzy search, we can make that breaking change as well 🫡

**Pull request merged**